### ISSUE TITLE
woke check fails for git submodules

### ISSUE THREAD

**Issue by fekete-robert** [get-woke/woke#192]
Hi, it seems that woke version 0.18.1 has a bug handling git submodules. I think the problem is with the fix to this woke issue (https://github.com/get-woke/woke/issues/102), causing woke to look for the .git/info/exclude directory, and throwing an error if it doesn't find it (instead of just moving on): 

```console
14-Apr [21:24:03.052]  Error: open .../.git/info/exclude: not a directory
```

The problem is that git submodules don't have a .git/ directory, only a .git file.

- [ x ] Yes, I've searched similar issues on GitHub and didn't find any.
- [ x ] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of woke</summary>

```console
$ woke --version
woke version 0.18.1
```

</details>

<details><summary>Config file</summary>

Default

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
# paste output here
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ woke --debug
2022-04-19T09:47:28+02:00 DBG woke version 0.18.1 built from 939e604 on 2022-03-19T15:45:29Z
2022-04-19T09:47:28+02:00 DBG no config file loaded, using only default rules
2022-04-19T09:47:28+02:00 DBG default rules rules=["whitelist","blacklist","master-slave","slave","grandfathered","man-hours","sanity","dummy","guys","whitebox","blackbox"]
2022-04-19T09:47:28+02:00 DBG all enabled rules rules=["whitelist","blacklist","master-slave","slave","grandfathered","man-hours","sanity","dummy","guys","whitebox","blackbox"]
2022-04-19T09:47:28+02:00 DBG finished compiling ignores durationMS=148.256037
2022-04-19T09:47:28+02:00 DBG woke completed durationMS=148.479722
Error: open /Users/.../.git/info/exclude: not a directory
Usage:
  woke [globs ...] [flags]

Flags:
  -c, --config string           Config file (default is .woke.yaml in current directory, or $HOME)
      --debug                   Enable debug logging
      --disable-default-rules   Disable the default ruleset
      --exit-1-on-failure       Exit with exit code 1 on failures
  -h, --help                    help for woke
      --no-ignore               Ignored files in .gitignore, .ignore, .wokeignore, .git/info/exclude, and inline ignores are processed
  -o, --output string           Output type [text,simple,github-actions,json,sonarqube] (default "text")
      --stdin                   Read from stdin
  -v, --version                 version for woke

2022-04-19T09:47:28+02:00 FTL  error="open /Users/.../.git/info/exclude: not a directory"

```

</details>

**Comment by caitlinelfring** [comment]
Thanks for reporting! My guess this really has something to do with https://github.com/get-woke/woke/pull/117, which changed how ignores are handled. Can you do a few things for me?

* Do you get the same behavior with `v0.17.0`? You can install it with `curl -sSfL https://git.io/getwoke | bash -s -- -b ./ v0.17.0` and then run `./woke`?
* Would you mind sharing your `.gitmodules` or an example that produces the behavior so I can reproduce it? 

Thanks!

**Comment by fekete-robert** [comment]
Hi Caitlin, thanks for the reply.
woke version 0.17.1 works fine. Version 0.18.1 also finishes without errors when using the --no-ignores flag.
I'll create a test repo and get back to you.

**Comment by fekete-robert** [comment]
git clone https://github.com/fekete-robert/woke-test.git --recursive

**Comment by caitlinelfring** [comment]
I have a fix ready for this issue here: https://github.com/get-woke/woke/pull/194

**Comment by caitlinelfring** [comment]
This will be released in [`v0.18.2`](https://github.com/get-woke/woke/releases/tag/v0.18.2)
